ui/src/app/views/resource-card/component.tsx

```diff
@@ -113,175 +113,313 @@ export class ResourceCardMenuComponent extends ReactWrapperBase {
+  cohort: {
+    backgroundColor: '#F8C954'
+  },
+  conceptSet: {
+    backgroundColor: '#AB87B3'
+  },
+  notebook: {
+    backgroundColor: '#8BC990'
+  }
+});
+
+export interface ResourceCardProps {
+  resourceCard: RecentResource;
+  onUpdate: Function;
+}
+
+export interface ResourceCardState {
   resourceType: ResourceType;
-  @Input('resourceCard')
   resourceCard: RecentResource;
-  @Input('cssClass')
-  cssClass: string;
-  @Output() onUpdate: EventEmitter<void | NotebookRename> = new EventEmitter();
-  wsNamespace: string;
-  wsId: string;
-  resource: any;
-  router: Router;
-  invalidResourceError = false;
-  confirmDeleting = false;
-
-  renaming = false;
-  editing = false;
-
-  constructor(
-    private cohortsService: CohortsService,
-    private workspacesService: WorkspacesService,
-    private conceptSetsService: ConceptSetsService,
-    private signInService: SignInService,
-    private route: Router,
-  ) {}
-
-  ngOnInit() {
-    this.wsNamespace = this.resourceCard.workspaceNamespace;
-    // RW-1298 this should be updated
-    this.wsId = this.resourceCard.workspaceFirecloudName;
-    if (this.resourceCard) {
-      if (this.resourceCard.notebook) {
-        this.resourceType = ResourceType.NOTEBOOK;
-      } else if (this.resourceCard.cohort) {
-        this.resourceType = ResourceType.COHORT;
-      } else if (this.resourceCard.conceptSet) {
-        this.resourceType = ResourceType.CONCEPT_SET;
+  renaming: boolean;
+  editing: boolean;
+  confirmDeleting: boolean;
+  invalidResourceError: boolean;
+}
+
+export class ResourceCard extends React.Component<ResourceCardProps, ResourceCardState> {
+
+  constructor(props: ResourceCardProps) {
+    super(props);
+    const defaultState = {
+      editing: false,
+      renaming: false,
+      confirmDeleting: false
+    };
+    if (props.resourceCard) {
+      if (props.resourceCard.notebook) {
+        this.state = {
+          resourceType: ResourceType.NOTEBOOK,
+          resourceCard: props.resourceCard,
+          invalidResourceError: false,
+          ...defaultState
+        };
+      } else if (props.resourceCard.cohort) {
+        this.state  = {
+          resourceType: ResourceType.COHORT,
+          resourceCard: props.resourceCard,
+          invalidResourceError: false,
+          ...defaultState
+        };
+      } else if (props.resourceCard.conceptSet) {
+        // TODO [1/29/19]: Need to thread generated/fetch model types through for API calls
+        //  once parent components (notebook-list, cohort-list, concept-set-list) are converted
+        //  this should go away and resourceCard can be used as what comes through the props
+        const myTempConceptSet = {...this.props.resourceCard.conceptSet,
+          domain: props.resourceCard.conceptSet.domain as Domain};
+        this.state = {
+          resourceType: ResourceType.CONCEPT_SET,
+          resourceCard: {...props.resourceCard, conceptSet: myTempConceptSet},
```

Comment:
Both `resourceType` and `resourceCard` can be computed directly from props, and therefore don't need to be in state. (Even the temporary hack, which could just be factored out into e.g. a function if needed)

Reply:
even with this hack to cast the types in the case of a `ConceptSet`?  could you explain more what you mean?

Reply:
nevermind actually figured it out!  will push up soon